pkg/bff/collaborators.go

```diff
@@ -69,27 +94,198 @@ func (s *Server) ReplaceCollaborator(c *gin.Context) {
+		collaborator.CreatedAt = collaborator.ModifiedAt
+	}
+
 	// Save the updated organization
 	if err = s.db.UpdateOrganization(org); err != nil {
 		log.Error().Err(err).Msg("could not save organization with new collaborator")
-		c.JSON(http.StatusInternalServerError, "could not replace collaborator")
+		c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not replace collaborator"))
 		return
 	}
 
 	c.JSON(http.StatusOK, collaborator)
 }
+
+// DeleteCollaborator deletes the collaborator in the request from the user's
+// organization. The user must have the update:collaborators permission.
+// Note: This does not return an error if the collaborator does not exist on the
+// organization and instead returns a 200 OK response.
+func (s *Server) DeleteCollaborator(c *gin.Context) {
+	var (
+		err          error
+		collaborator *models.Collaborator
+		org          *models.Organization
+	)
+
+	// Fetch the organization from the claims
+	// NOTE: This method handles the error logging and response
+	if org, err = s.OrganizationFromClaims(c); err != nil {
+		return
+	}
+
+	// Get the collabID from the URL
+	collabID := c.Param("collabID")
+	if collabID == "" {
+		log.Warn().Msg("missing ID in delete collaborator request")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse("ID is required in order to delete a collaborator"))
+		return
+	}
+
+	// Make sure a 404 is returned if the collaborator does not exist
+	var ok bool
+	if collaborator, ok = org.Collaborators[collabID]; !ok {
+		log.Warn().Str("collabID", collabID).Msg("collaborator does not exist")
+		c.JSON(http.StatusNotFound, api.ErrorResponse("collaborator not found"))
+		return
+	}
+
+	// If the collabroator is already verified in Auth0, then remove them from the
+	// organization
+	if collaborator.VerifiedAt != "" && collaborator.UserId != "" {
+		// Fetch the user from Auth0
+		var user *management.User
+		if user, err = s.auth0.User.Read(collaborator.UserId); err != nil {
+			log.Error().Err(err).Str("collabID", collabID).Str("auth0_id", collaborator.UserId).Msg("could not fetch user from Auth0")
+			c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not fetch user from Auth0"))
+			return
+		}
+
+		// Fetch the user app metadata
+		appdata := &auth.AppMetadata{}
+		if err = appdata.Load(user.AppMetadata); err != nil {
+			log.Error().Err(err).Str("collabID", collabID).Str("auth0_id", collaborator.UserId).Msg("could not parse user app metadata")
+			c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not parse user app metadata"))
+			return
+		}
+
+		// Update the app metadata with the removed organization
+		appdata.OrgID = ""
+		if err = s.SaveAuth0AppMetadata(*user.ID, *appdata); err != nil {
+			log.Error().Err(err).Str("collabID", collabID).Str("auth0_id", collaborator.UserId).Msg("could not save user app metadata")
+			c.JSON(http.StatusInternalServerError, api.ErrorResponse("could not save user app metadata"))
+			return
+		}
```

Comment:
We should check to see what happens when the user tries to log in without an OrgID -- I think they'll get an error right?

Reply:
Hmm, I guess what would happen is that they would just get assigned a new OrgID. In other words the Login method doesn't currently distinguish between a new user and a user that has been removed from an org.

Reply:
Oh good point - so probably not an issue!